core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -982,6 +676,100 @@ class PartKeyLuceneIndex(ref: DatasetRef,
   }
 }
 
+protected class LuceneQueryBuilder extends PartKeyQueryBuilder {
```

Comment:
This takes the existing inline object creation from the query builder and wraps it to the builder pattern.  The idea is that we maintain a stack of BooleanQueries to support nesting.  As non-boolean queries are visited they are added to the top of the stack.  The end result should be a single top level BooleanQuery that is popped off at the end.

This is heavily exercised in existing unit tests that run queries so I chose not to duplicate more tests, but I'm interested in feedback on that.